msal/authority.py

```diff
@@ -109,13 +110,16 @@ def __init__(
                     % authority_url)
             tenant_discovery_endpoint = payload['tenant_discovery_endpoint']
         else:
-            tenant_discovery_endpoint = (
-                'https://{}:{}{}{}/.well-known/openid-configuration'.format(
-                    self.instance,
-                    443 if authority.port is None else authority.port,
-                    authority.path,  # In B2C scenario, it is "/tenant/policy"
-                    "" if tenant == "adfs" else "/v2.0" # the AAD v2 endpoint
-                    ))
+            if is_ciam and len(authority.path) > 1:
+                raise ValueError(
+                    "CIAM authority should look like https://contoso.ciamlogin.com")
```

Comment:
it's also ok to have
- contoso.ciamlogin.com/contoso.onmicrosoft.com
- contoso.ciamlogin.com/GUID of contoso.onmicrosoft.com
- contoso.ciamlogin.com/contoso.com (if contoso.com is a valid domain of the contoso.onmicrosoft.com tenant)

Reply:
OK, I have switched the implementation and added [these 2 new test cases to cover path-less and with-path scenarios](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/543/files#diff-1982837bc801e709175e25839e1046a445df689d7a2e6309ccb6856e350dee7eR89-R99). Please review again, @jmprieur